File: programming 1/lab 6 kotlin/src/main/java/server/commands/UpdateCommand.java
package server.commands;

import common.communication.Response;
import server.managers.CollectionManager;

import java.io.BufferedReader;
import java.io.IOException;

/**
 * Команда для обновления элемента коллекции по его идентификатору.
 */
public class UpdateCommand extends Command implements Promptable{
    private final CollectionManager cm;
    public UpdateCommand(CollectionManager cm) {
        super("update id", "обновить элемент по id");
        this.cm = cm;
    }

    /**
     * Выполняет команду.
     * @param id @param id идентификатор элемента.
     */
    @Override
    public Response execute(String id) {
        if (id == null) throw new IllegalArgumentException("Команда update требует аргумент id\n");
        return cm.update(Long.parseLong(id));
    }

    /**
     * Выполняет команду из файла.
     * @param reader для чтения данных из файла.
     * @param id идентификатор элемента.
     */
    @Override
    public Response execute(BufferedReader reader, String id) throws IOException {
        if (id == null) throw new IllegalArgumentException("Команда update требует аргумент id\n");
        try {
            return cm.update(reader, Long.parseLong(id));
        } catch (NumberFormatException e) {
            throw new NumberFormatException("Неверный формат аргумента! Требуется числовое значение.\n");
        }
    }
}

File: programming 1/lab 6/kotlin/lab 6/src/main/java/server/commands/UpdateCommand.kt
package server.commands

import common.communication.Response
import server.managers.CollectionManager
import java.io.BufferedReader
import java.io.IOException

/**
 * Команда для обновления элемента коллекции по его идентификатору.
 */
class UpdateCommand(private val cm: CollectionManager) : Command("update id", "обновить элемент по id"), Promptable {
    /**
     * Выполняет команду.
     * @param id @param id идентификатор элемента.
     */
    override fun execute(id: String?): Response {
        requireNotNull(id) { "Команда ${name.split(' ')[0]} требует аргумент ${name.split(' ')[1]}" }
        return cm.update(id.toLong())
    }

    /**
     * Выполняет команду из файла.
     * @param reader для чтения данных из файла.
     * @param id идентификатор элемента.
     */
    @Throws(IOException::class)
    override fun execute(reader: BufferedReader, id: String?): Response {
        requireNotNull(id) { "Команда ${name.split(' ')[0]} требует аргумент ${name.split(' ')[1]}" }
        try {
            return cm.update(reader, id.toLong())
        } catch (e: NumberFormatException) {
            throw NumberFormatException("Неверный формат аргумента! Требуется числовое значение.\n")
        }
    }
}
